examples/dcgan/dcgan.gop

```diff
@@ -28,13 +30,108 @@ var (
 	cuda       = flag.Bool("cuda", false, "enables cuda")
 	dryRun     = flag.Bool("dry-run", false, "check a single training cycle works")
 	ngpu       = flag.Int("ngpu", 1, "number of GPUs to use")
-	netG       = flag.String("netG", "", "path to netG (to continue training)")
-	netD       = flag.String("netD", "", "path to netD (to continue training)")
+	netGPath   = flag.String("pathG", "", "path to netG (to continue training)")
+	netDPath   = flag.String("pathD", "", "path to netD (to continue training)")
 	outf       = flag.String("outf", ".", "folder to output images and model checkpoints")
 	manualSeed = flag.Int("manualSeed", 0, "manual seed")
 	classes    = flag.String("classes", "bedroom", "comma separated list of classes for the lsun data set")
 )
 
+
+type Generator struct {
+	ngpu int
+	main torch.Module
+}
+
+func NewGenerator(gpu int, nz int, ngf int, nc int) *Generator {
+	return &Generator {
+		ngpu: gpu,
+		main: torch.Sequential(
+			[
+				// input is Z, going into a convolution
+				torch.ConvTranspose2d(nz, ngf * 8, 4, 1, 0, false),
+				torch.BatchNorm2d(ngf * 8)
+				torch.ReLU(true),
+				// state size. (ngf*8) x 4 x 4
+				torch.ConvTranspose2d(ngf * 8, ngf * 4, 4, 2, 1, false),
+				torch.BatchNorm2d(ngf * 4),
+				torch.ReLU(true),
+				// state size. (ngf*4) x 8 x 8
+				torch.ConvTranspose2d(ngf * 4, ngf * 2, 4, 2, 1, false),
+				torch.BatchNorm2d(ngf * 2),
+				torch.ReLU(true),
+				// state size. (ngf*2) x 16 x 16
+				torch.ConvTranspose2d(ngf * 2, ngf, 4, 2, 1, false)
+				torch.BatchNorm2d(ngf),
+				torch.ReLU(true),
+				// state size. (ngf) x 32 x 32
+				torch.ConvTranspose2d(ngf, nc, 4, 2, 1, false),
+				torch.Tanh()
+			]
+		),
+	}
+}
+
+func (self Generator) Forward(x torch.Tensor) output torch.Tensor {
+	if x.IsCuda && self.ngpu > 1 {
+		output = torch.DataParallel(self.main, x, range(self.ngpu))
+	} else {
+		output = self.main.Forward(x)
+	}
+}
+
+type Discriminator struct {
+	ngpu int
+	main torch.Module
+}
+
+func NewDiscriminator(gpu int, ndf int, nc int) *Discriminator {
+	return &Discriminator{
+		ngpu: gpu,
+		main: torch.Sequential(
+			[
+				// input is (nc) x 64 x 6
+				torch.Conv2d(nc, ndf, 4, 2, 1, false),
+				torch.LeakyReLU(0.2, true),
+				// state size. (ndf) x 32 x 32
+				torch.Conv2d(ndf, ndf * 2, 4, 2, 1, false),
+				torch.BatchNorm2d(ndf * 2),
+				torch.LeakyReLU(0.2, true),
+				// state size. (ndf*2) x 16 x 16
+				torch.Conv2d(ndf * 2, ndf * 4, 4, 2, 1, false),
+				torch.BatchNorm2d(ndf * 4),
+				torch.LeakyReLU(0.2, true),
+				// state size. (ndf*4) x 8 x 8
+				torch.Conv2d(ndf * 4),
+				torch.BatchNorm2d(ndf * 8),
+				torch.LeakyReLU(0.2, true),
+				// state size. (ndf*8) x 4 x 4
+				torch.Conv2d(ndf * 8, 1, 4, 1, 0, true),
+				torch.Sigmoid()
+			]
+		),
+	}
+}
+
+func (self Discriminator) Forward(x torch.Tensor) output torch.Tensor {
```

Comment:
The receiver here should be pointer type to avoid copying the value before calling the method.

Reply:
Done, fix it.